libpius/signer.py

```diff
@@ -33,10 +33,14 @@ class PiusSigner(object):
   GPG_SIG_BEG = '[GNUPG:] BEGIN_SIGNING'
   GPG_SIG_CREATED = '[GNUPG:] SIG_CREATED'
 
-  def __init__(self, signer, mode, keyring, gpg_path, tmpdir, outdir,
-               encrypt_outfiles, mail, mailer, verbose, sort_keyring,
+  def __init__(self, signer, force_signer, mode, keyring, gpg_path, tmpdir,
+               outdir, encrypt_outfiles, mail, mailer, verbose, sort_keyring,
                policy_url, mail_host):
     self.signer = signer
+    self.force_signer = force_signer
+    self.forced_signer = self.signer
```

Comment:
I thought about that but that has an unintended side effect: When you use the '!' for signer only this key (and no other subkeys) is copied to the temp keyring. As this key probably only has the sign capability and not the encrypt capability it is not a valid recipient for use in -r 

Also another probably more versatile variant would be to have --force-signer be a keyid like --signer so that could do something like --signer 0x12345 and --force-signer 0x34567! which would use 0x12345 as recipient (and probably other operations) and 0x34567! for signing.